### DISCUSSION TITLE
API Key copy to clipboard icon.

### DISCUSSION THREAD

**Discussion by Zorbadil** [magicbug/Cloudlog#3011] [General]
Is it just me. or do the icons to copy to clipboard on the API Key page not work?

**Comment by phl0** [comment]
> Is it just me. or do the icons to copy to clipboard on the API Key page not work?

Are you using HTTP or HTTPS?

**Reply by Zorbadil** [reply]
HTTP  I'm running cloudlog locally on a  raspberry pi.

**Reply by phl0** [reply]
Then it's probably blocked by your browser.

**Reply by Zorbadil** [reply]
I don't think so.  I've tried two seperate browsers, and they allowed me to copy the commands I used on the installation instructions when I installed Cloudlog (V2.66).

I'm having doubts that Cloudlog has fully installed correctly, although no errors have been thrown up during the install, and I have imported my log4om2 logbook in adif format (after altering the PHP settings to allow me to upload a 5MB log).

**Comment by magicbug** [comment]
The copy function will only work for HTTPS its a limitation of the web browser and also some browsers wont even support the function you can read more about it at https://developer.mozilla.org/en-US/docs/Web/API/Clipboard

You can of course see any errors by hitting F12 on the keyboard and looking at the console within the developer tools area, with HTTP only I get a nice error.

**Reply by Zorbadil** [reply]
Is it supposed to run in HTTPS?  Have I done something wrong in the installation do you think?  I tried the dietpi installation, but it installs version 8.2 of PHP, and this leads to errors in the installation and running of Cloudlog.  

To get what I thought was a successful installation I manually installed LAMP, but apart from configuring PHP to allow me to import a 5mb ADIF, and obviously I followed your setup guide to create the database and grant privileges, other than that I have done nothing.

Is there a setup I need to follow to get it to run in HTTPS?  Apologies but I only have a rudimentary working knowlege of linux.

Nik.

**Comment by HB9HCI** [comment]
BTW: if I use the CL-copy function I get a API like c**I**...4711. But the value seen on display is c**1**...4711. If I hit the test-button, CL sends an URL with .../c**I**...4711 with result `valid`. If I change the URL to .../c**1**...4711 the result is `Key Invalid - either not found or disabled`.
weird!
Andreas, HB9HCI
